### PR TITLE
feat: play audio in sequence

### PR DESCRIPTION
**Pull request by reinaldonetof**
## How to test or reproduce

- Test audios in sequence
- Pls, see the video at # Screenshots to see the edge cases that I was able to avoid

## Screenshots

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/47038980/93720aa7-a2bc-4cdb-97bf-2fadb114f633

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

[TC-1095]

[TC-1095]: https://rocketchat.atlassian.net/browse/TC-1095?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/containers/AudioPlayer/index.tsx [hunk 1]
```diff
@@ -134,16 +136,21 @@ const AudioPlayer = ({
...
 		return () => {
 			unsubscribeFocus();
+			unsubscribeBlur();
 		};
 	}, [navigation]);
 
 	useEffect(() => {
 		const listener = EventEmitter.addEventListener(AUDIO_FOCUSED, ({ audioFocused }: { audioFocused: string }) => {
-			setFocused(audioFocused === audioUri.current);
+			setFocused(!!audioFocused && audioFocused === audioUri.current);
```
Comment (reinaldonetof) [inline app/containers/AudioPlayer/index.tsx:153]: This fixes a visual bug created by the focused state when the audio isn't downloaded yet.

File: app/lib/methods/audioPlayer.ts [hunk 2]
```diff
@@ -14,17 +22,30 @@ const AUDIO_MODE = {
...
+
 class AudioPlayer {
 	private audioQueue: { [audioKey: string]: Audio.Sound };
 	private audioPlaying: string;
+	private audiosRendered: Set<string>;
 
 	constructor() {
 		this.audioQueue = {};
 		this.audioPlaying = '';
+		this.audiosRendered = new Set<string>();
```
Comment (reinaldonetof) [inline app/lib/methods/audioPlayer.ts:35]: This property holds the values of the audios displayed on the screen. Consequently, if there is another audio in sequence but it hasn't been rendered, the audio won't play automatically. This adjustment was made to address a scenario where you have two audios in the main room. If the last audio is preceded by another audio, located within a thread that you've opened before, the issue is now resolved.

Comment (jsathu07) [reply app/lib/methods/audioPlayer.ts:35]: Great catch @reinaldonetof If you come across any issues with my contribution, please feel free to let me know. I'm always available and more than happy to contribute. Additionally, if it's convenient for you, ping me in official react native rc channel , that way I can clarify my doubts in real time, Thanks

Comment (reinaldonetof) [reply app/lib/methods/audioPlayer.ts:35]: Hey @jsathu07 first of all, thanks a lot for your initiative and PR. 🚀 

Regarding the changes and some refactors that I made, don't worry about them. After we complete the internal review and QA process, I will discuss with you to merge these changes into your PR. Then, you will be able to merge it into develop.

**Review by diegolmello — APPROVED** [review]

**Comment by diegolmello** [comment]
@jsathu07 Hey, thanks for contributing! I changed it a little bit and triggered CI. Going to test and merge Tomorrow 🤞🏼!

**Comment by jsathu07** [comment]
Thank you @diegolmello  for considering my pull request. I'm looking forward to contribute more
